testing/test_skipping.py

```diff
@@ -153,6 +154,127 @@ def test_func(self):
         assert skipped
         assert skipped.reason == "condition: config._hackxyz"
 
+    def test_skipif_markeval_namespace(self, testdir):
+        testdir.makeconftest(
+            """
+            import pytest
+
+            def pytest_markeval_namespace():
+                return {"color": "green"}
+            """
+        )
+        p = testdir.makepyfile(
+            """
+            import pytest
+
+            @pytest.mark.skipif("color == 'green'")
+            def test_1():
+                assert True
+
+            @pytest.mark.skipif("color == 'red'")
+            def test_2():
+                assert True
+        """
+        )
+        res = testdir.runpytest(p)
+        assert res.ret == 0
+        res.stdout.fnmatch_lines(["*1 skipped*"])
+        res.stdout.fnmatch_lines(["*1 passed*"])
+
+    def test_skipif_markeval_namespace_multiple(self, testdir):
```

Comment:
@nicoddemus, I added the suggested scenario, however, root skips while the other two don't.

How are `ihooks` evaluated?
Top level conftest then lower level conftest? That's how I'm expecting it to work but the test scenario is proving me wrong...